apps/extension/server/src/server.ts

```diff
@@ -202,8 +208,105 @@ connection.onCodeLens(({ textDocument }) => {
 const areTypesWritable = (typegenData: TypegenData) =>
   deepEqual(typegenData.typesNode.value, typegenData.data.tsTypesValue);
 
+// this shouldn't use file system directly but rather an injected module
+// atm this extension doesn't work in browsers anyway though
+async function isXState5(uri: string) {
+  try {
+    const pkgJsonPath = require.resolve('xstate/package.json', {
+      paths: [uriToFileName(UriUtils.dirname(URI.parse(uri)).toString())],
+    });
+    return (require(pkgJsonPath).version as string).startsWith('5');
```

Comment:
What if it's like `"^5.4.1"`?

Reply:
It can’t be. I’m checking XState’s package.json and its version here - not a dependency range used to install XState in the project

Reply:
Ahh okay.